test/distributed/org/apache/cassandra/distributed/UpgradeableCluster.java

```diff
@@ -42,6 +42,7 @@ private UpgradeableCluster(Builder builder)
 
     protected IUpgradeableInstance newInstanceWrapper(int generation, Versions.Version version, IInstanceConfig config)
     {
+        config.set(Constants.KEY_DTEST_API_CONFIG_CHECK, false);
```

Comment:
It seems we always want to turn off check for jvm dtest. 
In case we want to have test cases to check config, unit test could be more useful. 
So maybe we can just hardcode it in `Instance.java`?

Reply:
> It seems we always want to turn off check for jvm dtest.

Nope!  They were added to block stuff like typos (turning this on actually found typos in tests!) and to match the behavior of CassandraDaemon.  This got complicated in upgrade as the dtest-api isn't that good about knowing the instance's version and adapting to it (python dtest is better with this at the moment).

So, my feeling is that normal tests should have this on by default, but to disable in upgrade tests as the jvm-dtest api makes it hard to do that correctly.